src/AppInstallerCommonCore/UserSettings.cpp

```diff
@@ -84,12 +84,56 @@ namespace AppInstaller::Settings
             return {};
         }
 
+        template <Setting S>
+        std::optional<typename details::SettingMapping<S>::policy_t> GetValueFromPolicy(int)
+        {
+            // return std::nullopt;
+            return GroupPolicies().GetValue<details::SettingMapping<S>::Policy>();
+        }
+
+        template <Setting S>
+        std::optional<typename details::SettingMapping<S>::json_t> GetValueFromPolicy(long)
+        {
+            using T = decltype(std::declval<details::SettingMapping<S>::json_t>());
+            return std::nullopt;
+        }
+
+        template <Setting S>
+        std::optional<typename details::SettingMapping<S>::json_t> GetValueFromPolicy()
+        {
+            return GetValueFromPolicy<S>(0);
+        }
+
         template <Setting S>
         void Validate(
             Json::Value& root,
             std::map<Setting, details::SettingVariant>& settings,
             std::vector<std::string>& warnings)
         {
+            // Settings set by Group Policy override anything else. See if there is one.
+            auto policyValue = GetValueFromPolicy<S>();
+            if (policyValue.has_value())
+            {
+                // If the value is valid, use it.
+                // Otherwise, fall back to the settings.
```

Comment:
I think we actually have to have "if the policy is valid, use it.  otherwise fall back to the default".  If an admin has configured a value, I don't think we can use the user value ever.